core/src/main/java/org/elasticsearch/client/transport/TransportClientNodesService.java

```diff
@@ -431,11 +431,11 @@ public void run() {
                                     // if its one of the actual nodes we will talk to, not to listed nodes, fully connect
                                     if (nodes.contains(listedNode)) {
                                         logger.trace("connecting to cluster node [{}]", listedNode);
-                                        transportService.connectToNode(listedNode);
+                                        transportService.connectToNodeLight(listedNode);
```

Comment:
Same. The comment implies this will "fully" connect rather than "light" connect.